JetNews/app/src/main/java/com/example/jetnews/ui/AppDrawer.kt

```diff
@@ -44,34 +45,43 @@ import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import androidx.navigation.compose.KEY_ROUTE
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
 import com.example.jetnews.R
 
 @Composable
 fun AppDrawer(
-    navigateTo: (Screen) -> Unit,
-    currentScreen: Screen,
+    navController: NavController,
```

Comment:
I think (we need more experience on this) that passing navController down is a smell.

Ideally, everything related to navigation is encapsulated in the NavHost file.

I think this can be reverted and the navController stuff hoisted

Reply:
Done